plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -28,17 +44,97 @@
  */
 public class GoogleLoginService {
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final String STASH_OAUTH_CRED_KEY = "OAUTH_CRED";
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email", //$NON-NLS-1$
+      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
   ));
 
   /**
-   * Returns the credential of the active user. If there is no active user, returns {@code null}.
+   * Returns the credential of an active user (among multiple logged-in users). A login screen
+   * may be presented, e.g., if no user is logged in or login is required due to an expired
+   * credential.
+   *
+   * @param shellProvider provides a shell for the login screen if login is necessary
+   * @throws IOException can be thrown by the underlying Login API request
+   */
+  // TODO(chanseok); synchronize properly
```

Comment:
> This has always bothered me. It seems like virtually all Eclipse (and Intellij too) code doesn't care about synchronization.

There are other avenues for synchronization than just the `synchronize` keyword.
  * A lot of the code we use are implicitly synchronized by only being able to be called from the UI/SWT thread. That includes most of the Eclipse Workbench (IWorkbench and friends) and the E4 Modelled Workbench (`getTransientData()` etc).  They can sometimes be read from other threads, but they can only be modified on the UI thread.  
  * `getService()` is backed by `IEclipseContext`s which are thread-safe.
  * Other things are assumed to be protected by running in Jobs which has scheduling/locking support.